### PR TITLE
Borgs are now more explosion resistant and properly

### PR DESCRIPTION
**Pull request by ZWRh3**
## About the PR 
makes all mobs use the `PROP_MOB_EXPLOPROT` property when calculating explosion resistance

makes cyborgs use base their explosion damage on the power argument of ex_act rather than the severity
## Why's this needed? 
currently cyborgs are extremely weak to explosions and the 3 possible values severity can have are very limiting

also makes all mobs use the PROP_MOB_EXPLOPROT property properly

changes the force shield upgrade to halve explosion damage

fire shield now only protects erebite cells

NOTE: cyborg shield upgrades should probably get merged into one shield upgrade but that is out of scope of this PR

### CODE REVIEW

**Comment by ZWRh3** [comment]
!merge_upstream

**Comment by ZWRh3** [comment]
!merge_upstream

**Review by Tarmunora — CHANGES_REQUESTED** [review]
anything with an adjusted power >= 6 should gib outright
phys shield should not modify severity if it also gives exploprot

**Comment by ZWRh3** [comment]
> anything with an adjusted power >= 6 should gib outright phys shield should not modify severity if it also gives exploprot

severity is the fallback it gets both severity and power and onlu uses power if it gets it (some things like wrestlers directly call explosions without power but actual explosions dont)

**Comment by ZWRh3** [comment]
> anything with an adjusted power >= 6 should gib outright phys shield should not modify severity if it also gives exploprot

like any and all borgs from light to heavy?

**Comment by Tarmunora** [comment]
> > phys shield should not modify severity if it also gives exploprot
> 
> severity is the fallback it gets both severity and power and onlu uses power if it gets it (some things like wrestlers directly call explosions without power but actual explosions dont)

in the fallback case, power is set based on severity, and is *then* reduced by exploprot.

> > anything with an adjusted power >= 6 should gib outright
> like any and all borgs from light to heavy?

Yes. Power 6 after adjustment must gib outright.

**Review by Tarmunora — APPROVED** [review]

File: code/mob/living/silicon/robot.dm [hunk 1]
```diff
@@ -827,38 +827,50 @@
...
+			switch(severity)
+				if(1)
+					power = 9
+				if(2)
+					power = 5
+				if(3)
+					power = 3
+		power *= clamp(1-src.get_explosion_resistance(), 0, 1)
+		if (power >= 6)
+			SPAWN(1 DECI SECOND)
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot.dm:857]: why

File: code/mob/living/silicon/robot.dm [hunk 2]
```diff
@@ -827,38 +827,50 @@
...
+		/*switch(severity)
+			if(1.0)
 				SPAWN(1 DECI SECOND)
 					src.gib(1)
 				return
-			if(2) damage = 40
-			if(3) damage = 20
+			if(2.0) damage = 40
+			if(3.0) damage = 20
+		*/
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot.dm:869]: commented code :(

File: code/mob/living/silicon/robot.dm [hunk 3]
```diff
@@ -827,38 +827,50 @@
...
-				boutput(src, "<span class='notice'>Your fire shield absorbs some of the blast!</span>")
-				playsound(src.loc, "sound/impact_sounds/Energy_Hit_1.ogg", 40, 1)
-				fire_protect = 1
-				severity++
-
-		var/damage = 0
-		switch(severity)
-			if(1)
+				boutput(src, "<span class='notice'>Your fire shield absorbs the heat of the blast!</span>")
+				playsound(src, "sound/impact_sounds/Energy_Hit_1.ogg", 40, 1)
```
Comment (ZeWaka) [inline code/mob/living/silicon/robot.dm:844]: fix sound paths for code guidelines pls

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/mob/living/silicon/robot.dm (hunk 1 above)
Comment (ZWRh3) [reply code/mob/living/silicon/robot.dm:857]: humans do it too i thinks its so the explosion wont sleep

**Pull request merged**